src/models/project-settings.js

```diff
@@ -2,43 +2,51 @@ import path from 'path';
 import jf from 'jsonfile';
 import { pwd } from 'shelljs';
 import rc from 'rc';
+import cc from 'rc/lib/utils';
+import _zipObject from 'lodash/zipObject';
+import _map from 'lodash/map';
 
-/*
-  Look into using Yam for finding settings so it will get the first
-  .reduxrc it finds and use that for project settings just like how
-  eslintrc and ember-cli works.
-*/
-
-/*
-  2.0 TODO
-
-  Use rc to enable multiple .reduxrc files to be located and merged into a
-  single object.
-
-  rc: https://www.npmjs.com/package/rc
-*/
+import BlueprintCollection, {parseSetting as parseBlueprint} from './blueprint-collection';
 
 export default class ProjectSettings {
   // public & tested - maintain in 2.0
   constructor (defaultSettings = {}, args = null) {
     this.defaultSettings = defaultSettings;
     this.args = args;
+    this.blueprintChunks = [];
+    this.configChunks = [];
+    this.myParse = this.myParse.bind(this);
+    this.saveDefaults = this.saveDefaults.bind(this);
     this.loadSettings();
+    this.blueprints = new BlueprintCollection(_zipObject(this.configDirs(), this.blueprintChunks));
+  }
+
+  configDirs () {
+    return _map(this.configFiles(), (configFile) => path.dirname(configFile));
+  }
+
+  configFiles() {
+    return _map(this.settings.configs, (configFile) => path.resolve(configFile));
+  }
+
+  allConfigs() {
+    const configs = _zipObject(this.settings.configs, this.configChunks);
+    configs['__default__'] = this.defaultSettings;
+    return configs;
   }
 
   // internal & tested
   // from #constructor
   loadSettings () {
-    this.settings = rc('redux', this.defaultSettings, this.args);
-    // if the config file list is empty, and the config is only the default
-    // maybe save or prompt to save the default config file
+    const startingSettings = Object.assign({}, this.defaultSettings);
+    this.settings = rc('blueprint', startingSettings, this.args, this.myParse);
```

Comment:
Does rc  modify startingSettings? Might we have an issue with values in this.defaultSettings being modified since the values will be shared by startingSettings.

Reply:
Yes it does. And yes we will or at least might.

default settings aren't being used right now, but we are going to need to use them. We need the default settings to have whatever settings we need to have in order to run the CLI in the complete absence of .blueprintrc files.  It also needs to add our package blueprints directory.

The only real reason to maintain the integrity of the defaultSettings is to be able display just those settings, which is useful for troubleshooting complex configs.
if we keep it then change to: `startingSettings = JSON.parse(JSON.stringify(defaultSettings))`?

Reply:
Sure. Quick and easy. If it ever becomes problematic could migrate to a deep copy.